File: authorization/src/main/java/orwir/gazzit/authorization/model/Scope.kt
package orwir.gazzit.authorization.model

import java.util.*

enum class Scope {
    Identity,
    Edit,
    Flair,
    History,
    Modconfig,
    Modflair,
    Modlog,
    Modposts,
    Modwiki,
    Mysubreddits,
    Privatemessages,
    Read,
    Report,
    Save,
    Submit,
    Subscribe,
    Vote,
    Wikiedit,
    Wikiread;

    fun asParameter() = name.toLowerCase(Locale.ENGLISH)
}

File: lib/access/model/scope.dart
enum Scope {
  identity,
  edit,
  flair,
  history,
  modconfig,
  modflair,
  modlog,
  modposts,
  modwiki,
  mysubreddits,
  privatemessages,
  read,
  report,
  save,
  submit,
  subscribe,
  vote,
  wikiedit,
  wikiread,
}

extension ScopeExtensions on Scope {
  String get label => toString().split('.')[1];
  String get parameter => label;
}
